File: umbrella-insurance-core/src/main/java/com/umbrella/insurance/core/models/entities/Item.java
package com.umbrella.insurance.core.models.entities;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

@Getter
@Setter
@Entity
@Table(name = "items", schema = "public", uniqueConstraints = {
        @UniqueConstraint(name = "items_unique", columnNames = {"item_name"})
})
public class Item {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "item_id", nullable = false)
    private Long id;

    @NotNull
    @Column(name = "item_name", nullable = false, length = Integer.MAX_VALUE)
    private String itemName;

    @NotNull
    @Column(name = "price", nullable = false)
    private Double price;

    @NotNull
    @Column(name = "description", nullable = false, length = Integer.MAX_VALUE)
    private String description;

    @Column(name = "item_image")
    private byte[] itemImage;

    @NotNull
    @Column(name = "upc", nullable = false, length = Integer.MAX_VALUE)
    private String upc;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "unit_id")
    private Unit unit;

}

File: umbrella-insurance-react-ui/src/models/items/v1/Item.ts
import { Unit } from "../../units/v1/Unit";

export class Item {
    id?: number;
    itemName?: string;
    price?: number;
    description?: string;
    itemImage?: number[];
    upc?: string; 
    unit?: Unit;
    quantity?: number;
    constructor(obj?:any) {
        if(obj) {
            if(obj?.id) {
                this.id = obj.id;
            }
            if(obj?.itemName) {
                this.itemName = obj.itemName;
            }
            if(typeof obj?.price === 'number') {
                this.price = obj.price;
            }
            if(obj?.description) {
                this.description = obj.description;
            }
            if(obj?.itemImage) {
                this.itemImage = obj.itemImage;
            }
            if(obj?.upc) {
                this.upc = obj.upc;
            }
            if(obj?.unit) {
                this.unit = new Unit(obj.unit);
            }
            if(obj?.quantity) {
                this.quantity = obj?.quantity;
            }
        }
    }
    /*
    toJSON() {
        return {
            "id": this.id ? this.id?.toString() : null,
            "itemName": this.itemName ? this.itemName?.toString() : null,
            "price": this.price ? this.price : null,
            "description": this.description ? this.description : null,
            "itemImage": this.itemImage ? this.itemImage : null,
            "upc": this.upc ? this.upc?.toString() : null,
            "unit": this.unit ? this.unit?.toString(): null
        }
    }
        */
}